### ISSUE TITLE
Docker build issue

### ISSUE THREAD

**Issue by devlin016** [nomeata/bSpokeLight#4]
I would like to compile this app because I want linux native for my wheel lights. I get this error on docker build.
`Removing intermediate container c812b4a4384c
 ---> f2f45cca4e1a
Step 4/9 : COPY cabal.config *.cabal /home/
COPY failed: stat /var/lib/docker/tmp/docker-builder739520375/cabal.config: no such file or directory`

**Comment by nomeata** [comment]
The docker build was contributed by @akvadrako, so maybe he can help.

But it shouldn’t be so hard to build manually as per instructions. Don’t be shy trying `nix`, if you have the disk space, it’s worth it! (Although initially you might be put off by download times and seeing it build stuff; that’s usually a beginner hurdle).

Or use the the binary releases, of course.

**Comment by devlin016** [comment]
I tried building my from source using nix and I received another error I cant remember which let me try again. im running MX linux

**Comment by devlin016** [comment]
`nix-build` nix -A linux-exe
unpacking 'https://github.com/input-output-hk/hackage.nix/archive/9f5061167d467f62b25977c09c86e6cb18dee565.tar.gz'...

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
error: program 'tar' failed with exit code 2
(use '--show-trace' to show detailed location `information)`

**Comment by devlin016** [comment]
and with --show-trace i get
`nix-build nix -A linux-exe --show-trace
unpacking 'https://github.com/input-output-hk/hackage.nix/archive/9f5061167d467f62b25977c09c86e6cb18dee565.tar.gz'...

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
error: while evaluating 'bSpokeLight-exe' at /home/HP/bSpokeLight/nix/default.nix:4:21, called from /home/HP/bSpokeLight/nix/default.nix:59:15:
while evaluating anonymous function at /home/HP/bSpokeLight/nix/pkgs.nix:9:1, called from /home/HP/bSpokeLight/nix/default.nix:18:19:
while evaluating the attribute 'config.hsPkgs' at undefined position:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:336:9:
while evaluating the option `hsPkgs':
while evaluating the attribute 'isDefined' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:374:5:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:362:9:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:458:7:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:28, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:17:
while evaluating 'dischargeProperties' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:416:25, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:349:62:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:232:58:
while evaluating the attribute 'config.hsPkgs' at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/component-driver.nix:43:3:
while evaluating the attribute 'packages' at undefined position:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:336:9:
while evaluating the option `packages':
while evaluating the attribute 'isDefined' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:374:5:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:362:9:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:458:7:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:28, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:17:
while evaluating 'dischargeProperties' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:416:25, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:349:62:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:232:58:
while evaluating the attribute 'config.packages' at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/plan.nix:196:5:
while evaluating 'filterAttrs' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/attrsets.nix:115:23, called from /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/plan.nix:203:8:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/attrsets.nix:116:29, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/attrsets.nix:116:18:
while evaluating anonymous function at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/plan.nix:203:28, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/attrsets.nix:116:62:
while evaluating the attribute 'revision' at /home/HP/bSpokeLight/nix/plan.nix:76:9:
while evaluating the attribute 'hackage.configs' at undefined position:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:336:9:
while evaluating the option `hackage.configs':
while evaluating the attribute 'isDefined' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:374:5:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:362:9:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:458:7:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:28, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:17:
while evaluating 'dischargeProperties' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:416:25, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:349:62:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:232:58:
while evaluating the attribute 'config.configs' at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/hackage.nix:82:3:
while evaluating 'flip' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/trivial.nix:74:16, called from /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/modules/hackage.nix:39:5:
while evaluating the attribute 'hackage.db' at undefined position:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:75:45, called from undefined position:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:336:9:
while evaluating the option `hackage.db':
while evaluating the attribute 'isDefined' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:374:5:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:362:9:
while evaluating the attribute 'values' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:458:7:
while evaluating anonymous function at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:28, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:348:17:
while evaluating 'dischargeProperties' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:416:25, called from /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:349:62:
while evaluating the attribute 'value' at /nix/store/bvl3i194k0xyyf8qxp5203ym0pyy3sf2-source/lib/modules.nix:232:58:
while evaluating the attribute 'config.db' at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/package-set.nix:22:7:
while evaluating anonymous function at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/lib/fetch-external.nix:7:3, called from /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/default.nix:46:16:
while evaluating anonymous function at /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/lib/override-with.nix:3:11, called from /nix/store/rwmn07w8ll94kjfnazy185sl6rgcf5rh-source/lib/fetch-external.nix:11:5:
program 'tar' failed with exit code 2
`

**Comment by akvadrako** [comment]
There used to be a `cabal.config` in the base directory of the repo (see https://github.com/akvadrako/bSpokeLight) and it's not there anymore, which is causing the issue. I am not familiar enough with cabal to know if it's still needed.

**Comment by devlin016** [comment]
dropped that cabal.config in the base directory and get this error i believe the config needs adjustments

`cabal: Could not resolve dependencies:
trying: bSpokeLight-0.1 (user goal)
next goal: base (dependency of bSpokeLight-0.1)
rejecting: base-4.9.0.0/installed-4.9... (conflict: bSpokeLight =>
base==4.12.*)
rejecting: base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0,
base-4.8.1.0, base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0,
base-4.6.0.1, base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0,
base-4.4.0.0, base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1,
base-4.2.0.0, base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1
(constraint from non-upgradeable package requires installed instance)
Dependency tree exhaustively searched.
The command '/bin/sh -c cabal update     && cabal install --only-dependencies' returned a non-zero code: 1`

**Comment by nomeata** [comment]
That's usually the symptom of using the wrong version of GHC

**Comment by devlin016** [comment]
could someone show me a correct cabal.config? this is kind out of my scope. I never even heard of GHC until now.

**Comment by nomeata** [comment]
I’m still surprised by your nix error; it works here. It looks as if it couldn't download something…

I think building with anything but nix isn’t really well supported at the moment. Also, a `cabal.config` shouldn’t be needed.

Can't you use the released binary from https://github.com/nomeata/bSpokeLight/releases?

**Comment by devlin016** [comment]
@nomeata oh wow ID10T error now. i didnt check for any compiled binarys. what about the stcgal.py if I need it? will it just work from the repo?

**Comment by nomeata** [comment]
Yes, I think so

**Comment by devlin016** [comment]
after i installed stcgal from pip3 i was able to get everything to work thank you

**Comment by nomeata** [comment]
Cool! Let me know how it went and what you built with it :-)
